include/rmm/mr/cnmem_memory_resource.hpp

```diff
@@ -42,7 +42,7 @@ class cnmem_memory_resource final : public device_memory_resource {
    * @param initial_pool_size Size, in bytes, of the intial pool size. When
    * zero, an implementation defined pool size is used.
    *---------------------------------------------------------------------------**/
-  explicit cnmem_memory_resource(std::size_t initial_pool_size = 0) {
+  explicit cnmem_memory_resource(std::size_t initial_pool_size = 0, unsigned flags = 0) {
```

Comment:
The public constructor for `cnmem_memory_resource` should only expose `initial_pool_size`. 

Make a `protected` constructor that looks something like this:
```
enum class pool_options : bool { 
   MANAGED, ///< Uses managed memory for pool
   CUDA ///< Uses CUDA device memory for pool
};
cnmem_memory_resource( std::size_t initial_pool_size = 0, pool_options pool_type = pool_options::CUDA)
```

Then translate the `enum` to the proper flags to `cnmemInit`.

Reply:
You also need to make the destructor for `cnmem_memory_resource` to be `virtual` now since you're inheriting from it.

Reply:
Why do you want an enum when cnmemInit is already receiving flags?

Reply:
Because the cnmem flags are an implementation detail that should be abstracted away.

Reply:
thats a good point I created it but gave it type short in case some day in the future there are other types we want to handle